component/org.wso2.carbon.identity.dpop/src/main/java/org/wso2/carbon/identity/dpop/util/Utils.java

```diff
@@ -0,0 +1,130 @@
+
+import java.security.interfaces.ECPublicKey;
+import java.security.interfaces.RSAPublicKey;
+import java.text.ParseException;
+
+import javax.xml.namespace.QName;
+
+/**
+ * This class provides utility functions for dpop implementation.
+ */
+public class Utils {
+
+    public static JdbcTemplate getNewTemplate() {
+
+        return new JdbcTemplate(UmPersistenceManager.getInstance().getDataSource());
+    }
+
+    public static boolean readConfigurations() {
+
+        IdentityConfigParser configParser = IdentityConfigParser.getInstance();
+        OMElement oauthElem = configParser.getConfigElement(DPoPConstants.OAUTH_CONFIG_ELEMENT);
+        //Property to define dpop state
+        OMElement dpopConfigElement = oauthElem
+                .getFirstChildWithName(getQNameWithIdentityNS(DPoPConstants
+                        .DPOP_CONFIG_ELEMENT));
+
+        if (dpopConfigElement != null) {
+            OMElement dpopEnabled =
+                    dpopConfigElement.getFirstChildWithName(getQNameWithIdentityNS(DPoPConstants.DPOP_ENABLED));
+
+            if (dpopEnabled != null) {
+                return Boolean.parseBoolean(dpopEnabled.getText());
+            }
+        }
+        return false;
+    }
+
+    public static String getThumbprintOfKeyFromDpopProof(String dPopProof)
+            throws IdentityOAuth2Exception {
+
+        try {
+            SignedJWT signedJwt = SignedJWT.parse(dPopProof);
+            JWSHeader header = signedJwt.getHeader();
+            return getKeyThumbprintOfKey(header.getJWK().toString(), signedJwt);
+        } catch (ParseException | JOSEException e) {
+            throw new IdentityOAuth2ClientException(DPoPConstants.INVALID_DPOP_PROOF, DPoPConstants.INVALID_DPOP_ERROR);
+        }
+    }
+
+    private static String getKeyThumbprintOfKey(String jwk, SignedJWT signedJwt)
+            throws ParseException, JOSEException {
+
+        JWK parseJwk = JWK.parse(jwk);
+        boolean validSignature;
+        if (DPoPConstants.ECDSA_ENCRYPTION.equalsIgnoreCase(String.valueOf(parseJwk.getKeyType()))) {
+            ECKey ecKey = (ECKey) parseJwk;
+            ECPublicKey ecPublicKey = ecKey.toECPublicKey();
+            validSignature = verifySignatureWithPublicKey(new ECDSAVerifier(ecPublicKey), signedJwt);
+            if (validSignature) {
+                return computeThumbprintOfECKey(ecKey);
+            }
+        } else if (DPoPConstants.RSA_ENCRYPTION.equalsIgnoreCase(String.valueOf(parseJwk.getKeyType()))) {
+            RSAKey rsaKey = (RSAKey) parseJwk;
+            RSAPublicKey rsaPublicKey = rsaKey.toRSAPublicKey();
+            validSignature = verifySignatureWithPublicKey(new RSASSAVerifier(rsaPublicKey), signedJwt);
+            if (validSignature) {
+                return computeThumbprintOfRSAKey(rsaKey);
+            }
+        }
+        return null;
+    }
+
+    public static String computeThumbprintOfRSAKey(RSAKey rsaKey) throws JOSEException {
```

Comment:
Let's add method comments to public methods. Fix in other places as well

Reply:
Fixed with 1543f22[](https://github.com/kalanika/identity-oauth-addons/commit/1543f22f8b5a539e059fb66dba6a4975a3d41d68).